src/interpreter/explicitControlEvaluator.ts

```diff
@@ -1,7 +1,286 @@
 import { type Result, type Value } from './types/evaluationResults';
 import { Stack } from '../utils/stack';
-import { type ExplicitControlEvaluatorState } from './types/state';
-import { type BaseNode, type Program } from '../ast/types';
+import {
+  type AgendaItem,
+  type AgendaItemEvaluatorMapping,
+  type ExplicitControlEvaluatorState
+} from './types/interpreter';
+import {
+  type AssignmentExpression,
+  type BinaryExpression,
+  type BlockStatement,
+  type BreakStatement,
+  type CallExpression,
+  type Constant,
+  type ContinueStatement,
+  type DefaultStatement,
+  type DoWhileStatement,
+  type EmptyStatement,
+  type ExpressionStatement,
+  type ForStatement,
+  type FunctionDeclaration,
+  type GotoStatement,
+  type Identifier,
+  type IdentifierStatement,
+  type IfStatement,
+  type LogicalExpression,
+  type Program,
+  type ReturnStatement,
+  type StringLiteral,
+  type SwitchStatement,
+  type VariableDeclaration,
+  type WhileStatement
+} from '../ast/types';
+import {
+  allocateStackAddresses,
+  constructClosure,
+  getArgNumbers,
+  getBlockVariableDeclarationNames,
+  getExternalDeclarationNames
+} from './utils';
+import { VirtualMemory } from '../memory/virtualMemory';
+import {
+  type EnvironmentInstr,
+  type FunctionApplicationInstr,
+  type FunctionAssigmentInstr,
+  type FunctionMarkInstr,
+  type ResetInstr
+} from './types/instruction';
+import {
+  constructEnvironmentInstr,
+  constructFunctionApplicationInstr,
+  constructFunctionAssignmentInstr,
+  constructFunctionMarkInstr,
+  constructResetInstr
+} from './instruction';
+import { TextMemory } from '../memory/textMemory';
+import { constructMainCallExpression } from '../ast/constructors';
+import { InvalidFunctionApplication } from './errors';
+import {
+  constructGlobalEnvironment,
+  extendEnvironment,
+  getEnvironmentValue
+} from './environment';
+import { isEmptyStatement } from '../ast/typeGuards';
+
+const evaluators: AgendaItemEvaluatorMapping = {
```

Comment:
Would it be better to put the evaluators at the bottom of the file so that people will have an easier time reading this file?